automata/base/animation.py

```diff
@@ -0,0 +1,256 @@
+                    circle,
+                    Animate.default_init(manim.Circle)().surround(
+                        circle, buffer_factor=0.8
+                    ),
+                )
+                if node.attr["shape"].startswith("double")
+                else circle
+            )
+            self.add(self.shape)
+            self.label = Animate.default_init(manim.Text)(
+                node.name, font_size=float(node.attr["fontsize"])
+            )
+            self.add(self.label)
+        else:
+            raise ValueError(
+                f"Invalid node shape: {node.attr['shape']}. "
+                "Only 'point', 'circle' and 'doublecircle' are supported."
+            )
+        x, y = (float(pt) / _POINTS_IN_INCH for pt in node.attr["pos"].split(","))
+        self.set_x(x)
+        self.set_y(y)
+
+
+class _ManimEdge(manim.VGroup):
+    """
+    The `ManimEdge` class represents an `Edge` object in the `AGraph` object with manim.
+
+    The `ManimEdge` class is a `VGroup`. For the convenience, it's also added with:
+
+    Parameters
+    ----------
+    edge : VGroup
+        The curved arrow made with a series of `CubricBezier` curves objects and an
+        `Arrow` object.
+    label : Optional[Text]
+        The label on the edge, which is the symbol of the transition.
+    """
+
+    edge: manim.VGroup
+    label: manim.Text | None
+
+    def __init__(self, edge: pgv.Edge) -> None:
+        r"""
+        Parameters
+        ----------
+        edge : `pygraphviz.Edge`
+            which to based on to construct a FA transition.<br>
+            `edge.attr` may contain:
+            - 'arrowsize': %f
+            - 'fontsize': %f (not exists when 'label' not exists)
+            - 'label': str (may not exists)
+            - 'lp': '%f,%f'
+            - 'pos': 'e,%f,%f(\s+%f,%f)*'
+        """
+        super().__init__()
+        self.edge = self.__parse_spline(edge.attr["pos"].replace("\\\r", ""))
+        self.add(self.edge)
+        if label_text := edge.attr["label"]:
+            self.label = Animate.default_init(manim.Text)(
+                label_text, font_size=float(edge.attr["fontsize"])
+            )
+            x, y = (float(pt) / _POINTS_IN_INCH for pt in edge.attr["lp"].split(","))
+            self.label.set_x(x)
+            self.label.set_y(y)
+            self.add(self.label)
+
+    @staticmethod
+    def __parse_spline(edge_pos: str) -> manim.VGroup:
+        """
+        Convert the pos attribute of the edge which is a string of spline pattern.
+
+        Parameters
+        ----------
+        edge_pos : str
+            a str of spline pattern:
+            - spline = endp point (triple)+
+            - point = "%f,%f"
+            - endp = "e,"point
+            - triple = point point point
+        """
+        points = edge_pos.split()
+        control_points = tuple(
+            (*(float(pt) / _POINTS_IN_INCH for pt in point.split(",")), 0)
+            for point in points[1:]
```

Comment:
```suggestion
            for point in itertools.islice(points, 1, None)
```
Nit: Avoid making a copy.